pkg/github/github.go

```diff
@@ -0,0 +1,26 @@
+package github
+
+import (
+	"context"
+	"os"
+
+	"github.com/google/go-github/v52/github"
+	"golang.org/x/oauth2"
+)
+
+type RepositoryContentGetOptions = github.RepositoryContentGetOptions
+
+// GetGithubClient creates the required github client
+func GetGithubClient(ctx context.Context) github.Client {
```

Comment:
What's the difference between this and getGithubClient()? https://github.com/ddev/ddev/blob/18237c7dc49fa67b2be755964e19c5e9323e6f43/cmd/ddev/cmd/get.go#L367

Reply:
It's mostly the same. I've prepared a global factory function which should be used later everywhere. See also general feedback.

Reply:
Solved by https://github.com/ddev/ddev/pull/4915